tokio/src/io/driver/mod.rs

```diff
@@ -176,6 +176,8 @@ impl Driver {
             Err(e) => return Err(e),
         }
 
+        self.inner.waker.did_wake();
```

Comment:
I'm looking to revive this PR but I need more context on what this line actually is doing and what would happen if it were to be applied to non-espidf targets?

Reply:
Nevermind I think I understand and I've found a slightly more durable solution by using wake() to reregister which implicitly wakes the blocking poll.